makelib/golang.mk

```diff
@@ -241,21 +258,33 @@ go.vendor: $(DEP)
 	@$(DEP) ensure || $(FAIL)
 	@$(OK) dep ensure
 
+go.mod.verify:
+	@$(INFO) verify dependencies have expected content
+	@$(GO) mod verify || $(FAIL)
+	@$(OK) go modules dependencies verified
+
+go.mod.vendor:
+	@$(INFO) go mod vendor
+	@$(GO) mod vendor || $(FAIL)
+	@$(MAKE) go.writeable.mod.folder
+	@$(OK) go mod vendor
+
+# makes sure the pkg/mod folder is writable (otherwise it won't be deleted)
+go.writeable.mod.folder:
+	@echo 
+	@#check if go modules is enabled
+ifeq ($(GO111MODULE),on)
+	@test -d $(shell go env GOPATH)/pkg/mod && find $(shell go env GOPATH)/pkg/mod -type d -exec chmod 777 {} \; || echo 
```

Comment:
`$(shell go env GOPATH)` gets the default GOPATH (even when the GOPATH env variable is not set). Since we are removing the env variable for GOPATH because of go modules, this command gets the default installation of go modules.